### DISCUSSION TITLE
combine several sensors to one JSON

### DISCUSSION THREAD

**Discussion by bam80** [arendst/Tasmota#20395] [Q&A] [answered]
Hi,
I need to combine two sensors tele data to a one MQTT JSON.
Currently, the data from BLE sensor is transmitted in a separate JSON (to the same topic as the other temp sensor, though).

Also, I would like to publish particular sensor's data to two different topics.

Is something like this possible with Tasmota?

**Comment by barbudor** [comment, marked as answer]
BLE sensors work independently from Tasmota native sensors

You can write a bit of Berry to grab the values published by BLE sensors using a berry rule and then either publish them to another topic, or add them to the standard SENSOR message within a driver's `json_append()` method

**Reply by sfromis** [reply]
> Also, I would like to publish particular sensor's data to two different topics.

This is way outside of how the Tasmota sensor framework operates, which is about combining sensors into a single TelePeriod payload, where you can choose the frequency. This does not apply to BLE or Zigbee sensors, which are not controlled by Tasmota, but instead send data when they "feel like it".

But if you want to generate extra messages, you can create Berry code to pick up the standard payloads, and do what you like with the data, including doing one or more new publish.

**Reply by bam80** [reply]
Thanks, I found a [BLErry](https://github.com/tony-fav/tasmota-blerry) project I think I could use for this purpose.
This is the function of interest, I need to remove `tasmota.publish()` from here and instead move the stuff to the added `json_append()` as you described: 
https://github.com/tony-fav/tasmota-blerry/blob/ce6c2f05c0d96b8b7dbc0b38166a526e778299d8/blerry/blerry.be#L609
